Mage.Sets/src/mage/cards/c/CyclopeanTomb.java

```diff
@@ -0,0 +1,197 @@
+        
+
+        // {2}, {tap}: Put a mire counter on target non-Swamp land. That land is a Swamp for as long as it has a mire counter on it. Activate this ability only during your upkeep.
+        Ability ability = new ConditionalActivatedAbility(Zone.BATTLEFIELD, new AddCountersTargetEffect(CounterType.MIRE.createInstance()), new GenericManaCost(2), new IsStepCondition(PhaseStep.UPKEEP), "{2}, {T}: Put a mire counter on target non-Swamp land. That land is a Swamp for as long as it has a mire counter on it. Activate this ability only during your upkeep.");
+        ability.addCost(new TapSourceCost());
+        ability.addTarget(new TargetLandPermanent(filter));
+        ability.addEffect(new BecomeSwampEffect(Duration.Custom, false, true, "Swamp"));
+        this.addAbility(ability);
+        // When Cyclopean Tomb is put into a graveyard from the battlefield, at the beginning of each of your upkeeps for the rest of the game, remove all mire counters from a land that a mire counter was put onto with Cyclopean Tomb but that a mire counter has not been removed from with Cyclopean Tomb.
+        Effect effect = new CreateDelayedTriggeredAbilityEffect(new CycleDelayedTriggeredAbility());
+        effect.setText("at the beginning of each of your upkeeps for the rest of the game, remove all mire counters from a land that a mire counter was put onto with {this} but that a mire counter has not been removed from with {this}.");
+        this.addAbility(new PutIntoGraveFromBattlefieldSourceTriggeredAbility(effect));
+    }
+
+    public CyclopeanTomb(final CyclopeanTomb card) {
+        super(card);
+    }
+
+    @Override
+    public CyclopeanTomb copy() {
+        return new CyclopeanTomb(this);
+    }
+}
+
+class BecomeSwampEffect extends BecomesBasicLandTargetEffect {
+    
+    public BecomeSwampEffect(Duration duration, boolean chooseLandType, boolean loseOther, String... landNames) {
+        super(duration, chooseLandType, loseOther, landNames);
+        staticText = "That land is a Swamp for as long as it has a mire counter on it.";
+    }
+    
+    public BecomeSwampEffect(final BecomeSwampEffect effect) {
+        super(effect);
+    }
+    
+    @Override
+    public boolean apply(Layer layer, SubLayer sublayer, Ability source, Game game) {
+        Permanent land = game.getPermanent(this.targetPointer.getFirst(game, source));
+        if (land == null) {
+            // if permanent left battlefield the effect can be removed because it was only valid for that object
+            this.discard();
+        } else if (land.getCounters(game).getCount(CounterType.MIRE) > 0) {
+            // only if Mire counter is on the object it becomes a Swamp.
+            super.apply(layer, sublayer, source, game);
+        }
+        return true;
+    }
+    
+    @Override
+    public BecomeSwampEffect copy() {
+        return new BecomeSwampEffect(this);
+    }
+}
+
+class CycleDelayedTriggeredAbility extends DelayedTriggeredAbility {
```

Comment:
Cycle has a meaning in MtG.  If this class does not interact with Cycle, I'd rather see its name be a bit more verbose.